src/cargo/ops/cargo_add/mod.rs

```diff
@@ -449,6 +452,45 @@ fn resolve_dependency(
     Ok(dependency)
 }
 
+fn fuzzy_lookup(
+    dependency: &mut Dependency,
+    lookup: impl Fn(&str) -> CargoResult<Option<Dependency>>,
+    gctx: &GlobalContext,
+) -> CargoResult<Option<Dependency>> {
+    if let Some(rename) = dependency.rename() {
+        // If it has a rename only lookup that name - analogue to Dependency::toml_key()
+        return lookup(rename);
+    }
+
+    for name_permutation in [
+        dependency.name.clone(),
+        dependency.name.replace('-', "_"),
+        dependency.name.replace('_', "-"),
+    ]
+    .into_iter()
+    .dedup()
+    {
+        let Some(dep) = lookup(&name_permutation)? else {
+            continue;
+        };
+
+        if dependency.name != name_permutation {
+            // Ignore fuzzy matches with non-registry sources
```

Comment:
This comment is a re-phrasing of the code and isn't describing the motivation behind it.

Based on the test, my guess is
```suggestion
            // Mirror the fuzzy matching policy of `PackageRegistry::query_vec`
```

Reply:
Sounds good.

I believe referencing `PackageRegistry::query()` is more clear, because `query_vec()` is only defined on the trait.